plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchConfig.java

```diff
@@ -117,6 +125,85 @@ public ElasticsearchConfig setDefaultSchema(String defaultSchema)
         return this;
     }
 
+    @NotNull
+    public Duration getIndexMetaDataCacheTtl()
+    {
+        return indexMetaDataCacheTtl;
+    }
+
+    @Config("elasticsearch.index-metadata-cache-ttl")
+    public ElasticsearchConfig setIndexMetaDataCacheTtl(Duration indexMetaDataCacheTtl)
+    {
+        this.indexMetaDataCacheTtl = indexMetaDataCacheTtl;
+        return this;
+    }
+
+    @Min(0)
+    public long getIndexMetaDataCacheMaximumSize()
+    {
+        return indexMetaDataCacheMaximumSize;
+    }
+
+    @Config("elasticsearch.index-metadata-cache-maximum-size")
+    public ElasticsearchConfig setIndexMetaDataCacheMaximumSize(long indexMetaDataCacheMaximumSize)
+    {
+        this.indexMetaDataCacheMaximumSize = indexMetaDataCacheMaximumSize;
+        return this;
+    }
+
+    public boolean isUnionSchemaIndicesForAlias()
+    {
+        return unionSchemaIndicesForAlias;
+    }
+
+    @Config("elasticsearch.union-schema-indices-for-alias")
+    @ConfigDescription("Union schema indices for alias")
+    public ElasticsearchConfig setUnionSchemaIndicesForAlias(boolean unionSchemaIndicesForAlias)
+    {
+        this.unionSchemaIndicesForAlias = unionSchemaIndicesForAlias;
+        return this;
+    }
+
+    @Min(0)
+    public int getMaxNumberOfIndicesForAliasSchema()
+    {
+        return maxNumberOfIndicesForAliasSchema;
+    }
+
+    @Config("elasticsearch.max-number-of-indices-for-alias-schema")
+    @ConfigDescription("Maximum number of indices to include for alias schema")
+    public ElasticsearchConfig setMaxNumberOfIndicesForAliasSchema(int maxNumberOfIndicesForAliasSchema)
+    {
+        this.maxNumberOfIndicesForAliasSchema = maxNumberOfIndicesForAliasSchema;
+        return this;
+    }
+
+    public boolean isFailOnAliasSchemaMismatch()
+    {
+        return failOnAliasSchemaMismatch;
+    }
+
+    @Config("elasticsearch.fail-on-alias-schema-mismatch")
+    @ConfigDescription("Fail on alias schema mismatch")
+    public ElasticsearchConfig setFailOnAliasSchemaMismatch(boolean failOnAliasSchemaMismatch)
+    {
+        this.failOnAliasSchemaMismatch = failOnAliasSchemaMismatch;
+        return this;
+    }
+
+    public String getDefaultDataTypeForAliasSchemaMismatch()
+    {
+        return defaultDataTypeForAliasSchemaMismatch;
+    }
+
+    @Config("elasticsearch.default-datatype-for-alias-schema-mismatch")
+    @ConfigDescription("Default datatype for alias schema mismatch")
+    public ElasticsearchConfig setDefaultDataTypeForAliasSchemaMismatch(String defaultDataTypeForAliasSchemaMismatch)
```

Comment:
I still think we shouldn't expose this setting personally. IMO it should always be `text` as I don't expect much utility here. Casting to text is an easy button for folks to show data in case of a mismatch...it should  not be a solution to handle a lack of schema validation in the long term. I think leaving the code as is and just removing this method should suffice if we go this route. Curious to hear @hashhar and @martint think.

If we do expose it, we should create tests all possible casts.

Reply:
Makes sense. Let's wait to hear from others. Initially even I didn't think of exposing it. Lets comment out the contain for now. We could leverage in future in case if something like this arises.

Reply:
Agreed with Brian. For types which have an unknown type `keyword/text` makes sense but for fields having different types across indices we shouldn't be casting them to text.

Note that if someone wants to read such data they can expose both indices as separate tables with separate schema in Trino anyway.

I'll leave the decision to @martint / @bitsondatadev since they are more familiar with ES and I'm just speaking from general SQL perspective.

Reply:
So, originally I actually was on board with casting to a `text` value on mismatches, but I was only against setting that to other values (e.g. cast to a long on mismatch). However, it just occurred to me that casting to a string probably doesn't work for an `object` or `nested` among other types, only numbers and strings. I have an idea to address this though. 

We would need to wait for the PR #7373 to be merged, which allows us to set a mapping of "raw json" fields to mark a field to not be deserialized but rather keep the field and any nested field below in a json text format that can be parsed or just left alone as a `VARCHAR`.

We can still update the value in the mapping to `text` when there is a mismatch,  and then upon detecting a mapping type mismatch, we can generate an `asRawJson` entry in the `_meta` field of elasticsearch for all the mismatched fields. Then when we read from any of the indices, we will ignore the type of that field and any nested structure, and just deserialize that whole field and any nested fields to be returned as raw json.

Thoughts?

Reply:
@bitsondatadev I have already mentioned this as a limitation in the description of this PR. Query result definitely fails in such cases when rows contain values that can't be typecasted to `VARCHAR` in presto.

`Expected a string or numeric value for field 'object1' of type VARCHAR: {long1=112233, string1=string} [HashMap]`

We could merge with this limitation and raise a new issue to address it. @bitsondatadev You could get `asRawJson` working and then we could pick this limitation & handle fields with both primitive & object types. 

Thoughts?

Reply:
Addressed.